godot-ffi/src/lib.rs

```diff
@@ -237,56 +286,74 @@ fn print_preamble(version: GDExtensionGodotVersion) {
 
 /// # Safety
 ///
-/// The interface must have been initialised with [`initialize`] before calling this function.
+/// - Must be called from the main thread.
+/// - The interface must have been initialised with [`initialize`] before calling this function.
+/// - Must only be called once for each `api_level`.
 #[inline]
-pub unsafe fn load_class_method_table(api_level: ClassApiLevel) {
+pub unsafe fn load_class_method_table(api_level: InitLevel) {
     out!("Load class method table for level '{:?}'...", api_level);
     let begin = std::time::Instant::now();
 
     #[cfg(not(feature = "codegen-lazy-fptrs"))]
-    let mut string_names = StringCache::new(get_interface(), builtin_lifecycle_api());
+    // SAFETY: The interface has been initialized.
+    let interface = unsafe { get_interface() };
+
+    #[cfg(not(feature = "codegen-lazy-fptrs"))]
+    // SAFETY: The interface has been initialized.
+    let mut string_names = StringCache::new(interface, unsafe { builtin_lifecycle_api() });
 
     let (class_count, method_count);
     match api_level {
-        ClassApiLevel::Server => {
+        InitLevel::Core => {
+            // Currently we dont need to do anything in `Core`, this may change in the future.
+            class_count = 0;
+            method_count = 0;
+        }
+        InitLevel::Servers => {
             #[cfg(feature = "codegen-lazy-fptrs")]
-            {
+            // SAFETY: The interface has been initialized and this function hasn't been called before.
+            unsafe {
                 initialize_class_server_method_table(ClassServersMethodTable::load());
             }
             #[cfg(not(feature = "codegen-lazy-fptrs"))]
-            {
+            // SAFETY: The interface has been initialized and this function hasn't been called before.
+            unsafe {
                 initialize_class_server_method_table(ClassServersMethodTable::load(
-                    get_interface(),
+                    interface,
                     &mut string_names,
                 ));
             }
```

Comment:
Since the safety assertion is the same in both blocks, do you think we should transpose this to
```rs
// SAFETY: The interface has been initialized and this function hasn't been called before.
unsafe {
    #[cfg(feature = "codegen-lazy-fptrs")]
    {
        initialize_class_server_method_table(ClassServersMethodTable::load());
    }
    #[cfg(not(feature = "codegen-lazy-fptrs"))]
    {
        initialize_class_server_method_table(ClassServersMethodTable::load(
            interface,
            &mut string_names,
        ));
    }
}
```